woocommerce/woocommerce
# Fix broken checkout address forms layout

### Changes proposed in this Pull Request:





Previously, the flex-basis property was set to `calc(50% - #{$gap-smaller})`. This calculation was causing some fields to display at 100% width in Firefox, instead of the intended 50%. The issue is likely due to how Firefox interprets `calc()` in the `flex-basis` property when the container size is not a definite size. To resolve this, the `flex-basis` value has been adjusted to `calc(50% - 12px)`. This change reduced the base size of the flex item and ensuring the fields display correctly at 50% width in Firefox.

Closes #46836.




### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open an incognito window using Firefox.
2. Add a product to the cart.
3. Go to the checkout page with the Checkout block.
4. Verify that the first and last name fields are next to each other, both in the shipping and billing address forms.
5. Repeat the last step with other browsers, such as Chrome and Safari.

<table>
<tr>
<td valign="top">Before:
<br><br>
<img width="1277" alt="Screenshot 2024-05-03 at 14 27 09" src="https://github.com/woocommerce/woocommerce/assets/3323310/b8a47eb9-ed30-40f5-90f7-02a0486b5b7e">
</td>
<td valign="top">After:
<br><br>
<img width="1278" alt="Screenshot 2024-05-03 at 14 10 03" src="https://github.com/woocommerce/woocommerce/assets/3323310/bd62afd3-f741-449e-81ab-1083cac2ab62">
</td>
</tr>
</table>



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Fix broken checkout address forms layout

#### Comment 

</details>
